Composer/packages/server/src/models/asset/assetManager.ts

```diff
@@ -109,19 +153,21 @@ export class AssetManager {
       return undefined;
     }
     const boilerplate = ExtensionContext.extensions.baseTemplates[0];
-    const location = Path.join(boilerplate.path, 'scripts', 'package.json');
-    try {
-      if (fs.existsSync(location)) {
-        const raw = fs.readFileSync(location, 'utf8');
-        const json = JSON.parse(raw);
-        if (json && json.version) {
-          return json.version;
-        } else {
-          return undefined;
+    if (boilerplate.path) {
+      const location = Path.join(boilerplate.path, 'scripts', 'package.json');
+      try {
+        if (fs.existsSync(location)) {
+          const raw = fs.readFileSync(location, 'utf8');
```

Comment:
maybe avoid using sync version in case it's a big file to read?

Reply:
@hatpick Its a very small file basically  just a blank template. In addition, we read it once and inside the getter above it always picks the cached value after the first time

Reply:
I do think I can change it to async for reading it the first time. But its not a P0 IMO.

Reply:
Ok, my thinking is that since it is on the server side and potentially many users will hit the sync read, it may cause perf issues down the road.